packages/storybook-utils/src/events.ts

```diff
@@ -0,0 +1,777 @@
+import type { Events } from "vue";
+
+// prettier-ignore
+type EventName<T extends Event> = 
+  T extends ClipboardEvent ? "ClipboardEvent"
+: T extends WheelEvent ? "WheelEvent"
+: T extends PointerEvent ? "PointerEvent"
+: T extends DragEvent ? "DragEvent"
+: T extends MouseEvent ? "MouseEvent"
+: T extends CompositionEvent ? "CompositionEvent"
+: T extends FocusEvent ? "FocusEvent"
+: T extends KeyboardEvent ? "KeyboardEvent"
+: T extends TouchEvent ? "TouchEvent"
+: T extends AnimationEvent ? "AnimationEvent"
+: T extends TransitionEvent ? "TransitionEvent"
+: "Event";
+
+/**
+ * Removes a prefix from a string type.
+ * @example ```ts
+ * type Trimmed = TrimStart<"onclick", "on">;
+ * const trimmed: Trimmed = "click";
+ * ```
+ */
+type TrimStart<
+  TString extends string,
+  Prefix extends string,
+> = TString extends `${Prefix}${infer Base}` ? Base : TString;
+
+type EventDocMap = {
+  [Key in keyof Events]: {
+    constructor: Docs<EventName<Events[Key]>>;
+    event: Docs<Lowercase<TrimStart<Key, "on">>>;
+  };
+};
+
+/**
+ * The doc type ensures that a name has a valid URL that includes the name.
+ */
+type Docs<Name extends string> = {
+  name: Name;
+  url: `${string}${Name}${string}`;
+};
+
+export const EVENT_DOC_MAP: EventDocMap = {
```

Comment:
Nice TypeScript magic :D
Couldn't we map the event.name from the key? If I see correctly, the EVENT_DOC_MAP is always the event.name with the `on` prefix

Reply:
I meant the keys from our own `EVENT_DOC_MAP` :) So when we use it in the Storybook util, we could extract the event name from the key so we don't need to maintain them manually